contracts/tonomy/src/tonomy.cpp

```diff
@@ -497,4 +497,63 @@ namespace tonomysystem
           .send();
    }
 
+   void tonomy::staketokens(name account_name, asset quantity) {
+      require_auth(account_name);
+      eosio::check(quantity.amount > 0, "Stake quantity must be greater than zero.");
+      staking_allocations _staking(get_self(), get_self().value);
+
+      _staking.emplace(account_name, [&](auto& row) {
```

Comment:
need to add unbounded array protection, like we do here https://github.com/Tonomy-Foundation/Tonomy-Contracts/blob/master/contracts/vesting.tmy/src/vesting.tmy.cpp#L42

Reply:
more efficient implementation that iterating through all the records

`auto num_rows = std::distance(vesting_table.begin(), vesting_table.end());`